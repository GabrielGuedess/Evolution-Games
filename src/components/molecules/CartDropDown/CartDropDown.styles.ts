import media from 'styled-media-query';

import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    width: calc(100vw - (1.6rem * 2));
    padding: 2rem 1.2rem;
    border-radius: ${theme.border.radius.regular};
    background: ${theme.colors.gameDetails};

    ${media.greaterThan('medium')`
      width: 45rem;
      position: relative;
    `}
  `}
`;

export const HeadingWrapper = styled.div`
  ${({ theme }) => css`
    padding: 1rem 0rem;
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin: 0.5rem 0;
    border-bottom: 1px solid ${theme.colors.whiteText};
  `}
`;

export const Heading = styled.h1`
  ${({ theme }) => css`
    color: ${theme.colors.whiteText};
    font-size: 1.4rem;
  `}
`;

export const ViewAll = styled.a`
  ${({ theme }) => css`
    color: ${theme.colors.primary};
    font-size: 1.2rem;
    padding: 1rem;
    text-decoration: underline;
    cursor: pointer;
  `}
`;

export const TotalAllWrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: space-between;
    margin-top: 2.5rem;

    strong {
      text-transform: uppercase;
      font-size: 1.6rem;
      color: ${theme.colors.whiteText};

      ${media.greaterThan('medium')`
        font-size: 1.8rem;
      `}
    }

    h2 {
      text-transform: uppercase;
      font-size: 1.6rem;
      color: ${theme.colors.whiteText};

      ${media.greaterThan('medium')`
        font-size: 1.8rem;
      `}
    }
  `}
`;
