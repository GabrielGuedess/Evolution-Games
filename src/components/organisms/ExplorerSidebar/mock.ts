const filterCategories = {
  title: 'Genre',
  name: 'genre',
  type: 'checkbox',
  fields: [
    { title: 'Action-adventure', name: 'action-adventure' },
    { title: 'Adventure', name: 'adventure' },
    { title: 'Sports', name: 'sports' },
    { title: 'Puzzle', name: 'puzzle' },
    { title: 'Horror', name: 'horror' },
    { title: 'Platform', name: 'platform' },
    { title: 'Fantasy', name: 'fantasy' },
    { title: 'RPG', name: 'role-playing' },
    { title: 'JRPG', name: 'jrpg' },
    { title: 'Simulation', name: 'simulation' },
    { title: 'Strategy', name: 'strategy' },
    { title: 'Shooter', name: 'shooter' },
  ],
};

const filterYear = {
  title: 'Year',
  name: 'year',
  type: 'checkbox',
  fields: [
    { title: '1990 - 2000', name: '1990-2000' },
    { title: '2001 - 2005', name: '2001-2005' },
    { title: '2006 - 2011', name: '2006-2011' },
    { title: '2012 - 2017', name: '2012-2017' },
    { title: '2018 - Now', name: '2018-now' },
  ],
};

const filterPlatforms = {
  title: 'Platforms',
  name: 'platforms',
  type: 'checkbox',
  fields: [
    {
      title: 'Playstation 4',
      name: 'ps4',
    },
    {
      title: 'Playstation 5',
      name: 'ps5',
    },
    {
      title: 'Xbox One',
      name: 'one',
    },
    {
      title: 'Xbox Series X',
      name: 'xs',
    },
  ],
};

const filterFeedback = {
  title: 'Notas',
  name: 'feedback',
  type: 'checkbox',
  fields: [
    { title: 'From 5.0', name: 'from-5.0' },
    { title: 'From 4.0', name: 'from-4.0' },
  ],
};

export const filterItems = [
  filterCategories,
  filterYear,
  filterPlatforms,
  filterFeedback,
];
