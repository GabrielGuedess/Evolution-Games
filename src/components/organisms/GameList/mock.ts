import { GameCardProps } from 'components/molecules/GameCard/GameCard';

const gameCardItems: GameCardProps[] = [
  {
    id: '1',
    title: 'God of War: Ragnar√∂k',
    genre: 'Action-adventure ',
    developer: 'Santa Monica Games',
    year: '2022',
    img: 'https://i.pinimg.com/originals/34/f8/e5/34f8e55845212171e8bb35edc5b08018.jpg',
    price: '$ 60.99',
    score: 5.0,
    platform: ['playstation'],
    slug: '',
    primaryColor: '#9db4d0',
  },
  {
    id: '2',
    title: 'Star Wars Jedi: Fallen Order',
    genre: 'Action-adventure ',
    developer: 'EA',
    year: '2019',
    img: 'https://cdn2.steamgriddb.com/file/sgdb-cdn/grid/935a281a70ae21d54479c9c468145f7e.png',
    price: '$ 30.00',
    score: 3.8,
    platform: ['playstation', 'pc', 'xbox'],
    slug: '',
    primaryColor: '#a8b6cb',
  },
  {
    id: '3',
    title: 'Cyberpunk 2077',
    genre: 'Action-adventure ',
    developer: 'CD PROJEKT RED',
    year: '2020',
    img: 'https://whatifgaming.com/wp-content/uploads/2022/05/Neon-Aesthetic.jpg',
    price: '$ 10.00',
    score: 1.8,
    platform: ['playstation', 'pc', 'xbox'],
    slug: '',
    primaryColor: '#dabbde',
  },
  {
    id: '4',
    title: 'The Last of Us Part II',
    genre: 'Action-adventure ',
    developer: 'Naughty Dog',
    year: '2020',
    img: 'https://www.jeuxvideo-live.com/wp-content/uploads/jvl/2022/07/le-multi-de-the-last-of-us-2-deevoile-quelques-secrets-de-conception-104207-large.jpg',
    price: '$ 60.99',
    score: 5.0,
    platform: ['playstation'],
    slug: '',
    primaryColor: '#b2a7a5',
  },
  {
    id: '5',
    title: 'Elden Ring',
    genre: 'Action-adventure ',
    developer: 'BANDAI NAMCO Entertainment Inc',
    year: '2022',
    img: 'https://image.api.playstation.com/vulcan/ap/rnd/202110/2000/aGhopp3MHppi7kooGE2Dtt8C.png',
    price: '$ 60.99',
    score: 4.8,
    platform: ['playstation'],
    slug: '',
    primaryColor: '#aeb195',
  },
  {
    id: '6',
    title: 'Horizon Forbidden West',
    genre: 'Action-adventure ',
    developer: 'Santa Monica Games',
    year: '2022',
    img: 'https://t.ctcdn.com.br/efyy2ADqd67kPvUfXBVJWMJiX_I=/1400x788/smart/filters:format(webp)/i499600.png',
    price: '$ 60.99',
    score: 4.8,
    platform: ['playstation'],
    slug: '',
    primaryColor: '#aeb1d2',
  },
  {
    id: '7',
    title: "Marvel's Spider-Man",
    genre: 'Action-adventure ',
    developer: 'Insomniac Games',
    year: '2018',
    img: 'https://image.api.playstation.com/vulcan/ap/rnd/202008/1020/b0OfuUs4lHGWG4VNlHcXwL5a.jpg',
    price: '$ 60.99',
    score: 4.7,
    platform: ['playstation', 'pc'],
    slug: '',
    primaryColor: '#9ab3c5',
  },
  {
    id: '8',
    title: 'The Witcher 3',
    genre: 'Action-adventure ',
    developer: 'CD PROJEKT SA',
    year: '2015',
    img: 'https://e.snmc.io/lk/f/x/1508cc32995d91b6206ee1b1e82b9338/8329790',
    price: '$ 60.99',
    score: 4.5,
    platform: ['playstation', 'pc', 'xbox'],
    slug: '',
    primaryColor: '#d1b2a3',
  },
];

export default gameCardItems;
