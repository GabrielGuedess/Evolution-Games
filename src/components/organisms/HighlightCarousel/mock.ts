import { HighlightGameProps } from 'components/molecules/HighlightGame/HighlightGame';

export const highlightMock: HighlightGameProps[] = [
  {
    img: '/img/games/starWarsHighlight.png',
    gameLogo: '/img/games/highlightName/nameStarWars.png',
    gameLogoAlt: 'Logo Star Wars',
    alt: 'Star Wars',
    releaseYear: new Date(2019, 10, 10),
    genre: 'Action-adventure',
    developer: 'Respawn Entertainment',
    platform: ['pc', 'playstation', 'xbox'],
  },
  {
    img: '/img/games/r8.jpg',
    gameLogo: '/img/games/highlightName/nameR8.png',
    gameLogoAlt: 'Logo Star Wars',
    alt: 'Resident Evil 8',
    releaseYear: new Date('09-10-2022'),
    genre: 'Action-adventure',
    developer: 'Respawn Entertainment',
    platform: ['pc', 'playstation', 'xbox'],
  },
  {
    img: '/img/games/starWars.jpg',
    gameLogo: '/img/games/highlightName/nameStarWars.png',
    gameLogoAlt: 'Logo Star Wars 2',
    alt: 'Star Wars 2',
    releaseYear: new Date('09-10-2022'),
    genre: 'Action-adventure',
    developer: 'Respawn Entertainment',
    platform: ['pc', 'playstation', 'xbox'],
  },
  {
    img: '/img/games/theLastOfUs.jpg',
    gameLogo: '/img/games/highlightName/nameTheLastOfUs.png',
    gameLogoAlt: 'Logo The last Of Us',
    alt: 'The last Of Us',
    releaseYear: new Date('09-10-2022'),
    genre: 'Action-adventure',
    developer: 'Respawn Entertainment',
    platform: ['playstation'],
  },
];
