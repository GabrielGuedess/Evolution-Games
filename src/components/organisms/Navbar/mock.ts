import { GameCartDropProps } from 'components/molecules/GameCartDrop/GameCartDrop';

export const cartItems: GameCartDropProps[] = [
  {
    id: 'the-witcher-3',
    src: '/img/games/theWitcher.jpg',
    genres: ['Action'],
    title: 'The Witcher 3',
    developer: 'CD Projekt',
    quantity: 1,
    platform: 'ps4',
    price: 100,
  },
  {
    id: 'gta-6',
    src: 'https://gmedia.playstation.com/is/image/SIEPDC/grand-theft-auto-5-hero-banner-desktop-01-ps5-en-04mar22?$2400px$',
    genres: ['Action'],
    title: 'GTA 6',
    developer: 'CD Projekt',
    quantity: 1,
    platform: 'ps4',
    price: 100,
  },
  {
    id: 'rainbow-six',
    src: 'https://image.api.playstation.com/vulcan/ap/rnd/202105/1919/6t4GYDWoLSZVeNjAJLTczMg8.jpg?w=1920',
    genres: ['Action'],
    title: 'Rainbow Six',
    developer: 'CD Projekt',
    quantity: 1,
    platform: 'ps4',
    price: 100,
  },
];
