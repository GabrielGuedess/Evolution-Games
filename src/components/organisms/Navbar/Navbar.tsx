import Link from 'next/link';
import { useRouter } from 'next/router';

import { useState } from 'react';

import * as NavigationMenu from '@radix-ui/react-navigation-menu';
import * as Popover from '@radix-ui/react-popover';
import { Squash as Hamburger } from 'hamburger-react';
import { Heart } from 'phosphor-react';

import { Logo } from 'components/atoms/Logo/Logo';
import MediaMatch from 'components/atoms/MediaMatch/MediaMatch';
import {
  AvatarUser,
  AvatarUserProps,
} from 'components/molecules/AvatarUser/AvatarUser';
import { CartButton } from 'components/molecules/CartButton/CartButton';
import { MenuMobile } from 'components/molecules/MenuMobile/MenuMobile';

import { CartItems } from './mock';
import * as S from './Navbar.styles';

export type MenuProps = {
  loading?: boolean;
} & AvatarUserProps;

export const Navbar = ({ loading = false, ...props }: MenuProps) => {
  const { asPath } = useRouter();
  const [isOpen, setIsOpen] = useState(false);

  return (
    <S.Wrapper>
      <Logo />

      <MediaMatch greaterThan="medium">
        <NavigationMenu.Root>
          <S.ListTextLink>
            <S.ItemTextLink>
              <Link href="/" passHref>
                <S.LinkTextLink active={asPath === '/'}>Home</S.LinkTextLink>
              </Link>
            </S.ItemTextLink>

            <S.ItemTextLink>
              <Link href="/games" passHref>
                <S.LinkTextLink active={asPath === '/games'}>
                  Explorer
                </S.LinkTextLink>
              </Link>
            </S.ItemTextLink>

            <S.ItemTextLink>
              <Link href="/games" passHref>
                <S.LinkTextLink active={asPath === '/games'}>
                  Contact
                </S.LinkTextLink>
              </Link>
            </S.ItemTextLink>

            <NavigationMenu.Indicator />
          </S.ListTextLink>

          <NavigationMenu.Viewport />
        </NavigationMenu.Root>
      </MediaMatch>

      {!loading && (
        <>
          <S.WrapperIcons>
            <Link href="/favorites" passHref>
              <S.HeartLink aria-label="Link Favorites">
                <Heart size={24} />
              </S.HeartLink>
            </Link>

            <CartButton items={CartItems} />

            <MediaMatch greaterThan="medium">
              <AvatarUser {...props} />
            </MediaMatch>
          </S.WrapperIcons>

          <MediaMatch lessThan="medium">
            <Popover.Root open={isOpen} onOpenChange={setIsOpen}>
              <Popover.Trigger>
                <Hamburger
                  size={23}
                  duration={0.5}
                  label="Show menu"
                  toggled={isOpen}
                  toggle={setIsOpen}
                />
              </Popover.Trigger>
              <Popover.Anchor />
              <MenuMobile path={asPath} {...props} />
            </Popover.Root>
          </MediaMatch>
        </>
      )}
    </S.Wrapper>
  );
};
