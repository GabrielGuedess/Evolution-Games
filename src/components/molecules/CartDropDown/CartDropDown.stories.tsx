import * as Popover from '@radix-ui/react-popover';
import { Story, Meta } from '@storybook/react/types-6-0';

import { cartItems } from 'components/organisms/Navbar/mock';

import { CartDropDownProps, CartDropDown } from './CartDropDown';

export default {
  title: 'Molecules/CartDropDown',
  component: CartDropDown,
  args: {
    items: cartItems,
  },
} as Meta<CartDropDownProps>;

export const Default: Story<CartDropDownProps> = args => (
  <Popover.Root>
    <CartDropDown {...args} />
  </Popover.Root>
);
