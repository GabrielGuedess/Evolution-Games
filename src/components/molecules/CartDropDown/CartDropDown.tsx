import { CloseButton } from 'components/atoms/CloseButton/CloseButton';
import {
  GameCartDrop,
  GameCartDropProps,
} from 'components/molecules/GameCartDrop/GameCartDrop';

import * as S from './CartDropDown.styles';

export type CartDropDownProps = {
  items: GameCartDropProps[];
};

export const CartDropDown = ({ items }: CartDropDownProps) => (
  <S.Wrapper>
    <CloseButton />

    <S.HeadingWrapper>
      <S.Heading>My Cart ({items.length})</S.Heading>

      <S.ViewAll href="/cart">View all</S.ViewAll>
    </S.HeadingWrapper>

    {items.map(game => (
      <GameCartDrop key={game.title} {...game} />
    ))}

    <S.TotalAllWrapper>
      <strong>Total</strong>
      <h2>
        {Intl.NumberFormat('pt-BR', {
          style: 'currency',
          currency: 'BRL',
        }).format(items.reduce((acc, cur) => cur.price + acc, 0))}
      </h2>
    </S.TotalAllWrapper>
  </S.Wrapper>
);
