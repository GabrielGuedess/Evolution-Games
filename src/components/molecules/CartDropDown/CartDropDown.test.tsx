import * as Popover from '@radix-ui/react-popover';
import { renderWithTheme } from 'utils/tests/helpers';

import { screen } from '@testing-library/react';

import { CartItems } from 'components/organisms/Navbar/mock';

import { CartDropDown } from './CartDropDown';

jest.mock('components/molecules/GameCartDrop/GameCartDrop', () => ({
  __esModule: true,
  GameCartDrop: function Mock({ children }: { children: React.ReactNode }) {
    return <div data-testid="Mock GameCartDrop">{children}</div>;
  },
}));

describe('<CartDropDown />', () => {
  it('should render the GameCartDrop', () => {
    // Arrange
    const { container } = renderWithTheme(
      <Popover.Root>
        <CartDropDown items={CartItems} />
      </Popover.Root>,
    );

    // Assert
    expect(screen.getAllByTestId('Mock GameCartDrop')).toHaveLength(
      CartItems.length,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
