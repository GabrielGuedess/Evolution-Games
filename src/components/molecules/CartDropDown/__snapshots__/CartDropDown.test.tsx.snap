// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<CartDropDown /> should render the GameCartDrop 1'] = `
.c1 {
  color: #855aff;
  padding: 0.5rem;
  -webkit-transition: color 0.3s linear;
  transition: color 0.3s linear;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  font-size: 1.2rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: absolute;
  top: 0.5rem;
  left: 2.1rem;
}

.c1 svg {
  cursor: pointer;
}

.c0 {
  width: calc(100vw - (1.6rem * 2));
  padding: 2rem 1.2rem;
  border-radius: 1rem;
  background: #010914;
}

.c2 {
  padding: 1rem 0rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0.5rem 0;
  border-bottom: 1px solid #F4F4F5;
}

.c3 {
  color: #F4F4F5;
  font-size: 1.4rem;
}

.c4 {
  color: #855aff;
  font-size: 1.2rem;
  padding: 1rem;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 2.5rem;
}

.c5 strong {
  text-transform: uppercase;
  font-size: 1.6rem;
  color: #F4F4F5;
}

.c5 h2 {
  text-transform: uppercase;
  font-size: 1.6rem;
  color: #F4F4F5;
}

@media (min-width:768px) {
  .c1 {
    left: 0.5rem;
  }
}

@media (min-width:768px) {
  .c0 {
    width: 45rem;
    position: relative;
  }
}

@media (min-width:768px) {
  .c5 strong {
    font-size: 1.8rem;
  }
}

@media (min-width:768px) {
  .c5 h2 {
    font-size: 1.8rem;
  }
}

<div
  class="c0"
>
  <button
    aria-label="Close Action"
    class="c1"
    type="button"
  >
    <svg
      fill="currentColor"
      height="20"
      viewBox="0 0 256 256"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        fill="none"
        height="256"
        width="256"
      />
      <line
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="16"
        x1="200"
        x2="56"
        y1="56"
        y2="200"
      />
      <line
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="16"
        x1="200"
        x2="56"
        y1="200"
        y2="56"
      />
    </svg>
  </button>
  <div
    class="c2"
  >
    <h1
      class="c3"
    >
      My Cart (
      3
      )
    </h1>
    <a
      class="c4"
      href="/cart"
    >
      View all
    </a>
  </div>
  <div
    data-testid="Mock GameCartDrop"
  />
  <div
    data-testid="Mock GameCartDrop"
  />
  <div
    data-testid="Mock GameCartDrop"
  />
  <div
    class="c5"
  >
    <strong>
      Total
    </strong>
    <h2>
      R$Â 300,00
    </h2>
  </div>
</div>
`;
